src/connector/src/source/kinesis/source/reader.rs

```diff
@@ -135,6 +136,18 @@ impl KinesisSplitReader {
             }
             match self.get_records().await {
                 Ok(resp) => {
+                    if resp.millis_behind_latest.is_none()
+                        && let Some(shard) = &resp.child_shards
+                        && !shard.is_empty()
+                    {
+                        // according to the doc https://docs.rs/aws-sdk-kinesis/latest/aws_sdk_kinesis/operation/get_records/struct.GetRecordsOutput.html
+                        //
+                        // > The list of the current shard's child shards, returned in the GetRecords API's response only when the end of the current shard is reached.
+                        //
+                        // It means the current shard is finished, ie. inactive, and we should stop reading it. Checking `millis_behind_latest` is a double check.
+                        // Other executors are going to read the child shards.
+                        finish_flag = true;
+                    }
```

Comment:
Sorry, this seems to be wrong. Did you manually tested it works? 


I just tested and found that no shards can finish (by adding a log at the end).

Specifically, it's the "double check" for `millis_behind_latest` being wrong.
- It's `None` only when response don't contain this field, which never happened in my test.
- It's `Some(0)` for active shards which don't have more new data.

For inactive shards:

The last time it has data
```
{
    "Records": [...], // non-empty
    "MillisBehindLatest": 0,
	"NextShardIterator": ...
	// no ChildShards
}
```
Then calling with the `NextShardIterator`
```
{
    "Records": [], // empty
    "MillisBehindLatest": 2665000, // non-zero
    "ChildShards": [...]
	// no NextShardIterator
```